site/pmguides/src/api_builder_and_postgres/api_builder_node_and_postgres.md

```diff
@@ -0,0 +1,528 @@
+
+## Create The Database
+Duration: 7
+
+Let's create the database for the ToDo items. This section assumes you've installed Postgres. As part of its base installation, it installs a shell application called **psql**. This may not be automatically added to your path. Search for it in your operating system's search prompt and run it. It will open a terminal window and ask for some information to connect to the server.
+
+Accept the defaults for `server`,`database`, `port`, and `username`. When you get to `password`, use the password you set for the **postgres** user in the previous section. 
+
+![psql login](./assets/psql_login.png)
+
+Run the following commands in the terminal window, one at a time. Remember that the semicolon is required at the end of the command.
+
+1. Create the database.
+```sql
+CREATE DATABASE todo WITH OWNER postgres;
+```
+
+2. Switch to using the database for further commands. Following this, the command prompt will change from `postgres=#` to `todo=#`. 
+```sql
+\c todo
+```
+
+3. Create a user for the API. Before you do this, the only other user is **postgres** and that is a superuser role with *all of the privileges* on the database. Creating a new role and scoping it just to the permissions required to do its work is always a good practice. 
+```sql
+CREATE USER todo_admin WITH PASSWORD 'leelu_dallas_multipass_6';
+```
+
+4. Create the database table to hold your ToDo items. This has three fields: 
+  - `id_code`: a unique key of (up 36 characters).
+  - `to_do`: the task to be done (up to 255 characters).
+  - `completed`: whether or not it's been completed (boolean).
+```sql
+CREATE TABLE todos (id_code varchar(36) NOT NULL UNIQUE, to_do varchar(255) NOT NULL, completed boolean);
+```
+
+5. Grant the required privileges on the `todos` table to the new user.
+```sql
+GRANT SELECT, INSERT, UPDATE ON todos TO todo_admin;
+```
+
+6. At this point, the scaffolding is done, but let's add two items to the database so there's something when you start sending requests.
+
+Item 1:
+```sql
+INSERT INTO todos (id_code, to_do, completed) VALUES ('todo1','Get something done', TRUE); 
+```
+Item 2:
+```sql
+INSERT INTO todos (id_code, to_do, completed) VALUES ('todo2','Get another thing done', FALSE);
+```
+
+7. Query the database to see the two items you added.
+```sql
+SELECT * FROM todos;
+```
+
+![psql SELECT query](./assets/psql_select.png)
+
+8. Quit out of the Postgres shell.
+```bash
+\q
+```
+
+Next, use OpenAPI to define an API.
+
+
+<!-- ------------------------ -->
+
+## Define the API
+Duration: 7
+
+A good way to describe an API is to use the [OpenAPI 3 standard](https://spec.openapis.org/oas/latest.html), which describes the API in a machine-readable format.
+
+Let's break up the description for better human readability and get an overview of how this is composed. 
+
+[Download the full version of this API specification](./assets/openapitest.yml) if you want to see it in its entirety.
```

Comment:
I don't see this file in the assets folder

Reply:
I moved it to a gist because the file was not loading in the local preview, but missed one of the links. Fixed with a link to the gist instead.